packages/graph/hash_graph/lib/graph/src/datastore/postgres/mod.rs

```diff
@@ -131,12 +258,83 @@ mod tests {
     async fn get_entity_types() -> Result<(), sqlx::Error> {
         let pool = PostgresDatabase::new(&DB_INFO).await?.pool;
 
-        let _rows: Vec<EntityType> = sqlx::query_as("SELECT * from entity_types")
+        let _rows: Vec<EntityTypeRow> = sqlx::query_as("SELECT * from entity_types")
             .fetch_all(&pool)
             .await
             .report()
             .attach_printable("Could not select entity types")?;
 
         Ok(())
     }
+
+    #[ignore]
+    #[tokio::test]
+    async fn create_data_type() -> Result<(), sqlx::Error> {
+        let db = PostgresDatabase::new(&DB_INFO).await?;
+
+        let account_id = create_account_id(&db.pool).await?;
+
+        db.create_data_type(
+            DataType::new(serde_json::json!({"hello": "world"})),
+            account_id,
+        )
+        .await
+        .expect("Could not create data type");
+
+        Ok(())
+    }
+
+    #[ignore]
+    #[tokio::test]
+    async fn get_data_type_by_identifier() -> Result<(), sqlx::Error> {
+        let db = PostgresDatabase::new(&DB_INFO).await?;
+
+        let account_id = create_account_id(&db.pool).await?;
+
+        let Qualified { id, inner, .. } = db
+            .create_data_type(
+                DataType::new(serde_json::json!({"hello": "world"})),
+                account_id,
+            )
+            .await
+            .expect("Could not create data type");
+
+        let data_type = db.get_data_type(id).await.expect("Could not get data type");
+
+        assert_eq!(data_type.inner, inner);
+
+        Ok(())
+    }
+
+    #[ignore]
+    #[tokio::test]
+    async fn update_existing_data_type() -> Result<(), sqlx::Error> {
+        let db = PostgresDatabase::new(&DB_INFO).await?;
+
+        let account_id = create_account_id(&db.pool).await?;
+
+        let data_type = db
+            .create_data_type(
+                DataType::new(serde_json::json!({"hello": "world"})),
+                account_id,
+            )
+            .await
+            .expect("Could not create data type");
```

Comment:
```suggestion
            .await?;
```

Reply:
Addressed in https://github.com/hashintel/hash/pull/734/commits/cb7380d5a924ca58377712f530db8280e82c3c5f (or a prior commit can't recall)